backend/editor/api.py

```diff
@@ -111,7 +111,7 @@ def file_cleanup(filepath):
     try:
         os.remove(filepath)
     except Exception as e:
-        raise HTTPException(status_code=500, detail="Taxonomy file not found for deletion") from e
+        raise HTTPException(status_code=404, detail="Taxonomy file not found for deletion") from e
```

Comment:
Not even sure if a HTTPException should be raised: 
- file_cleanup is used in a FastAPI background task so it ran after the HTTP response
- if `os.remove` raises the `FileNotFoundError` error, then the file is already cleaned up. I believe that logging the unexpected behaviour should be enough  

```suggestion
        except FileNotFoundError:
             log.warn(f"Taxonomy file {filepath} not found for deletion")  
        
```

Reply:
I agree it's ok to just log.